lib/ansible/modules/system/hostname.py

```diff
@@ -155,7 +155,6 @@ class GenericStrategy(object):
 
     def __init__(self, module):
         self.module = module
-        self.hostname_cmd = self.module.get_bin_path('hostname', True)
```

Comment:
keep this here, just add a attempt to get hostnamectl, then 'if present' you can use it over hostname, but fallback to current behaviour when it is not present, that way you keep backwards compatibility